src/hugr/validate.rs

```diff
@@ -997,16 +997,20 @@ mod test {
             .map_into()
             .collect_tuple()
             .unwrap();
-
+        let mut closure = b.infer_extensions().unwrap();
+        assert!(closure.remove(&copy).is_some());
         b.replace_op(
             copy,
             NodeType::pure(ops::CFG {
                 signature: FunctionType::new(type_row![BOOL_T], type_row![BOOL_T]),
             }),
         )
         .unwrap();
+        // We feed in the closure from above because we can't compute the closure
```

Comment:
This suggests we might want a `&mut Hugr` method that writes the values from the closure into the Hugr (Changing `input_extensions` from `None` to `Some`)

Reply:
Oh, hang on, that's `instantiate_extensions`